src/Service/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -183,6 +149,51 @@ public string GraphQLSchema()
             return _cosmosDb.GraphQLSchema;
         }
 
+        private void ParseSchemaGraphQLFieldsForGraphQLType()
+        {
+            string graphqlSchema = GraphQLSchema();
+            DocumentNode root = Utf8GraphQLParser.Parse(graphqlSchema);
+
+            IEnumerable<ObjectTypeDefinitionNode> objectNodes = root.Definitions.Where(d => d is ObjectTypeDefinitionNode).Cast<ObjectTypeDefinitionNode>();
+            foreach (ObjectTypeDefinitionNode node in objectNodes)
+            {
+                string typeName = node.Name.Value;
+                _graphQLTypeToFieldsMap.TryAdd(typeName, new List<FieldDefinitionNode>());
+                foreach (FieldDefinitionNode field in node.Fields)
+                {
+                    _graphQLTypeToFieldsMap[typeName].Add(field);
+                }
+            }
+        }
+
+        public List<string> GetSchemaGraphQLFieldNamesForEntityName(string entityName)
+        {
+            List<FieldDefinitionNode>? fields;
+            // Check if entity name is using alias name, if so, fetch graph type name with the alias name
+            foreach (string typeName in _graphQLSingularTypeToEntityNameMap.Keys)
+            {
+                if (_graphQLSingularTypeToEntityNameMap[typeName] == entityName && _graphQLTypeToFieldsMap.TryGetValue(typeName, out fields))
+                {
+                    return fields is null ? new List<string>() : fields.Select(x => x.Name.Value).ToList();
+                }
+            }
+
+            // Get all the fields for a graph type
+            _graphQLTypeToFieldsMap.TryGetValue(entityName, out fields);
+            return fields is null ? new List<string>() : fields.Select(x => x.Name.Value).ToList();
+        }
+
+        public string? GetSchemaGraphQLFieldTypeByFieldName(string graphQLType, string fieldName)
```

Comment:
Similar to the function above, shouldn't we first try to retrieve the actual graphqlType from the map _graphQLSingularTypeToEntityNameMap? 

Since the caller in GraphQLFilterParser calls it with the entityName which may not necessarily be the graphql singular type name: 

_metadataProvider.GetSchemaGraphQLFieldTypeByFieldName(queryStructure.EntityName, name);

Reply:
This method is to retrieve the GraphQL type, as suppose to retrieve the entity name. In the ``` GraphQLFilterParser```, the graphQL type to entity name mapping is done here. 
https://github.com/Azure/data-api-builder/blob/326bfd1454c26d2d02ed84ffcc8a3749bb2a43c2/src/Service/Models/GraphQLFilterParsers.cs#L198

Reply:
I understand its retrieving the GraphQL type of the field of the given entity. 

But to look up the fields map from the entity type name, we should first find the actual graphql type for the entity then do the lookup 

The caller here: [https://github.com/Azure/data-api-builder/blob/326bfd1454c26d2d02ed84ffcc8a3749bb2a43c2/src/Service/Models/GraphQLFilterParsers.cs#L188](https://github.com/Azure/data-api-builder/blob/326bfd1454c26d2d02ed84ffcc8a3749bb2a43c2/src/Service/Models/GraphQLFilterParsers.cs#L188) calls this function with the entity name. But if entity name is using alias, we should be doing the following check: 

` // Check if entity name is using alias name, if so, fetch graph type name with the alias name
            foreach (string typeName in _graphQLSingularTypeToEntityNameMap.Keys)
            {
                if (_graphQLSingularTypeToEntityNameMap[typeName] == graphQLType && _graphQLTypeToFieldsMap.TryGetValue(typeName, out fields))
                {
                    return fields is null ? new List<string>() : fields.Select(x => x.Name.Value).ToList();
                }
            }`

Reply:
I see what you meant now. Thank you very much for pointing this out, this would be an issue. I've fixed this and updated the code.